### DISCUSSION TITLE
Cross pollination with git-issue

### DISCUSSION THREAD

**Discussion by voronoipotato** [git-bug/git-bug#1365] [General]
This is a discussion issue about the differences between git-issue and git-bug, and what challenges or benefits have been found. They are both GPLv3 so it should be completely safe, and likely advantageous to do so, if just to avoid pitfalls and traps either has gone through. It would be useful for example to have some documentation in both repos whether they are compatible or convertible, and what the differences might be. This discussion task is not really intended to create more work for the creators (though they are welcome), but rather as a todo for us users/uninitiated to investigate and discuss.

https://github.com/dspinellis/git-issue/issues/97

**Comment by sudoforge** [comment]
hey @voronoipotato! i've recently joined as a maintainer and noticed that you closed this and `dspinellis/git-issue#97` earlier this year.

is this still something you'd like to explore and converse about?

it's late for me, so apologies for the brevity of the remarks below, but i'll try to enumerate some initial findings from a quick glance, as it appears there are very stark differences:

- go vs. posix-y shell
- git-bug has bridges to third party platforms (e.g. sync w/ github issues)
- git-bug provides three UIs: cli, tui, web vs git-issue's cli
- no, i don't believe the data models are interopable at all; git-issues appears to be using commit hashes (of `HEAD`) as identifiers; entities in `git-bug` are disassociated from commit identifiers (although it used to use sha-1 hashes, likely back when you asked this question)
- git-issues appears to store files in a repository; git-bug stores _objects_, and doesn't create files on your filesystem

that's all i have time for right now. if you or other users are interested in a more thorough comparison, i'll create an issue for diving deep into `git-issues` to understand it better than i can with the limited time i had tonight.

**Reply by voronoipotato** [reply]
Thanks for the attention! I think the work you did to write this response is a lot of what I was talking about. What did they decide differently and what was the rationale behind the differences, and is there things we could learn or take from it. I think free software flourishes most if we take eager advantage of the fact that we have access to other free software.
